From e3a6bd1f14d6115c8c4a690dce9f7756f3c0dbda Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@toradex.com>
Date: Mon, 10 Jun 2019 14:47:51 +0300
Subject: [PATCH 4/7] ARM: dts: colibri_vf: Add dcu0 node

Add dumb node for NXP Display Control Unit0(DCU), which permits DM_ENABLED
converted driver to be probed. Currently no display timings are provided
in this node.

Signed-off-by: Igor Opaniuk <igor.opaniuk@toradex.com>
---
 arch/arm/dts/vf-colibri-u-boot.dtsi | 4 ++++
 arch/arm/dts/vf-colibri.dtsi        | 5 +++++
 arch/arm/dts/vf.dtsi                | 6 ++++++
 3 files changed, 15 insertions(+)

diff --git a/arch/arm/dts/vf-colibri-u-boot.dtsi b/arch/arm/dts/vf-colibri-u-boot.dtsi
index db86739805..2294ee9551 100644
--- a/arch/arm/dts/vf-colibri-u-boot.dtsi
+++ b/arch/arm/dts/vf-colibri-u-boot.dtsi
@@ -21,3 +21,7 @@
 &uart0 {
 	u-boot,dm-pre-reloc;
 };
+
+&dcu0 {
+	u-boot,dm-pre-reloc;
+};
diff --git a/arch/arm/dts/vf-colibri.dtsi b/arch/arm/dts/vf-colibri.dtsi
index 91ca4e4ddd..9de4b28e87 100644
--- a/arch/arm/dts/vf-colibri.dtsi
+++ b/arch/arm/dts/vf-colibri.dtsi
@@ -14,6 +14,7 @@
 
 	aliases {
 		usb0 = &ehci0; /* required for ums */
+		display1 = &dcu0;
 	};
 
 	reg_usbh_vbus: regulator-usbh-vbus {
@@ -241,3 +242,7 @@
 	pinctrl-0 = <&pinctrl_uart0>;
 	status = "okay";
 };
+
+&dcu0 {
+	status = "okay";
+};
diff --git a/arch/arm/dts/vf.dtsi b/arch/arm/dts/vf.dtsi
index 5e3b2c5b9d..5f69d0fd6e 100644
--- a/arch/arm/dts/vf.dtsi
+++ b/arch/arm/dts/vf.dtsi
@@ -145,6 +145,12 @@
 				#gpio-cells = <2>;
 			};
 
+			dcu0: dcu@40058000 {
+				compatible = "fsl,vf610-dcu";
+				reg = <0x40058000 0x1200>;
+				status = "disabled";
+			};
+
 			ehci0: ehci@40034000 {
 				compatible = "fsl,vf610-usb";
 				reg = <0x40034000 0x800>;
-- 
2.13.6

